### PR TITLE
ORLP-230. Redirect from log.txt to orlp.log.

### PR DESCRIPTION
**Pull request by vldmr1703**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -30,28 +30,6 @@
...
-            <exclusions>
-                <exclusion>
-                    <artifactId>slf4j-log4j12</artifactId>
-                    <groupId>org.slf4j</groupId>
-                </exclusion>
-            </exclusions>
-        </dependency>
-
-        <dependency>
-            <groupId>org.springframework.boot</groupId>
```
Comment (antonbakalets) [inline pom.xml]: Why are you removing this?

File: src/main/java/com/softserve/academy/spaced/repetition/logger/LogException.java [hunk 2]
```diff
@@ -1,27 +0,0 @@
...
-
-import org.aspectj.lang.JoinPoint;
-import org.aspectj.lang.annotation.AfterThrowing;
-import org.aspectj.lang.annotation.Aspect;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Component;
-
-@Aspect
-@Component
-public class LogException {
```
Comment (antonbakalets) [inline src/main/java/com/softserve/academy/spaced/repetition/logger/LogException.java]: This class must stay, but use a different logger. Please read the description in the Jira case once again.

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -5,7 +5,7 @@ spring:
     password: 123456
     driver-class-name: com.mysql.jdbc.Driver
   jpa:
-    show-sql: true
```
Comment (antonbakalets) [inline src/main/resources/application.yml]: Please also update this property in TestDatabaseConfig.

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -43,5 +43,5 @@ app:
 logging:
   file: orlp.log
   level:
-    com.softserve.academy.spaced.repetition: DEBUG
```
Comment (antonbakalets) [inline src/main/resources/application.yml]: Please also increase unit test log level as following https://www.mkyong.com/spring-boot/spring-boot-test-how-to-stop-debug-logs/

File: src/main/java/com/softserve/academy/spaced/repetition/logger/LogException.java [hunk 5]
```diff
@@ -1,27 +1,27 @@
...
 import org.springframework.stereotype.Component;
+import org.slf4j.Logger;
 
 @Aspect
 @Component
 public class LogException {
-    @Autowired
-    Logger logger;
+
+    private static final Logger logger = LoggerFactory.getLogger(Application.class);
```
Comment (antonbakalets) [inline src/main/java/com/softserve/academy/spaced/repetition/logger/LogException.java]: The parameter value should be the class of this object.

Comment (antonbakalets) [inline src/main/java/com/softserve/academy/spaced/repetition/logger/LogException.java]: Static variable are named in upercase.

File: src/main/resources/logback-test.xml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration>
+    <include resource="org/springframework/boot/logging/logback/base.xml" />
```
Comment (antonbakalets) [inline src/main/resources/logback-test.xml]: You will need two loggers (as in application.yml):
```
<logger name="com.softserve.academy.spaced.repetition" level="info" />
<logger name="org.springframework" level="error" />
```

File: src/test/java/com/softserve/academy/spaced/repetition/config/TestDatabaseConfig.java [hunk 7]
```diff
@@ -38,7 +38,7 @@ public LocalContainerEntityManagerFactoryBean entityManagerFactory(DataSource da
 
         HibernateJpaVendorAdapter vendorAdapter = new HibernateJpaVendorAdapter();
         vendorAdapter.setDatabasePlatform("org.hibernate.dialect.H2Dialect");
-        vendorAdapter.setShowSql(true);
+        vendorAdapter.setShowSql(false);
```
Comment (antonbakalets) [inline src/test/java/com/softserve/academy/spaced/repetition/config/TestDatabaseConfig.java]: Good.

File: src/main/java/com/softserve/academy/spaced/repetition/logger/LogException.java (hunk 5 above)
Comment (antonbakalets) [reply src/main/java/com/softserve/academy/spaced/repetition/logger/LogException.java]: See com.softserve.academy.spaced.repetition.controller.CardController for example.

Comment (vldmr1703) [reply src/main/java/com/softserve/academy/spaced/repetition/logger/LogException.java]: Oh, sorry. I'll fix it.

File: src/main/resources/logback-test.xml (hunk 6 above)
Comment (vldmr1703) [reply src/main/resources/logback-test.xml]: For com.softserve.academy.spaced.repetition level="info" or level="error"?

Comment (antonbakalets) [reply src/main/resources/logback-test.xml]: It up to you. I suggest - info.

File: src/main/java/com/softserve/academy/spaced/repetition/logger/LogException.java [hunk 8]
```diff
@@ -1,27 +1,27 @@
...
-    @Autowired
-    Logger logger;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LogException.class);
 
     @AfterThrowing(value = "execution(* com.softserve.academy.spaced.repetition.controller.*.*(..))", throwing = "ex")
     public void loggingExceptionsFromControllers(JoinPoint joinPoint, Exception ex) {
-        System.out.println(logger);
-        logger.log("Exception in controller = " + joinPoint.getThis() + ", method = " + joinPoint.getSignature());
+        LOGGER.trace("Exception in controller = " + joinPoint.getThis() + ", method = " + joinPoint.getSignature());
```
Comment (antonbakalets) [inline src/main/java/com/softserve/academy/spaced/repetition/logger/LogException.java]: One more thing. 

- Exceptions should be logged at ERROR level.
- Sonar will report somethig like "...conditional formatting should be used..."
- the actual exception should be logged.

Please rewrite as 
`LOGGER.error("Exception in controller = {} method = {}", joinPoint.getThis(), joinPoint.getSignature(), ex);`

The same below.

File: src/main/java/com/softserve/academy/spaced/repetition/logger/LogException.java [hunk 9]
```diff
@@ -1,27 +1,27 @@
...
 import org.aspectj.lang.annotation.AfterThrowing;
 import org.aspectj.lang.annotation.Aspect;
-import org.springframework.beans.factory.annotation.Autowired;
+import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Component;
+import org.slf4j.Logger;
 
 @Aspect
 @Component
 public class LogException {
```
Comment (antonbakalets) [inline src/main/java/com/softserve/academy/spaced/repetition/logger/LogException.java]: Lets rename to LogExceptionAspect?

File: src/main/java/com/softserve/academy/spaced/repetition/logger/LogExceptionAspect.java [hunk 10]
```diff
@@ -3,25 +3,24 @@
...
-        System.out.println(logger);
-        logger.log("Exception in controller = " + joinPoint.getThis() + ", method = " + joinPoint.getSignature());
+        LOGGER.error("Exception in controller = {} method = {}", joinPoint.getThis(), joinPoint.getSignature(), ex);
     }
 
     @AfterThrowing(value = "execution(* com.softserve.academy.spaced.repetition.service.*.*(..))", throwing = "ex")
     public void loggingExceptionsFromServices(JoinPoint joinPoint, Exception ex) {
-        System.out.println(logger);
-        logger.log("Exception in service = " + joinPoint.getThis() + ", method = " + joinPoint.getSignature());
+        LOGGER.error("Exception in controller = {} method = {}", joinPoint.getThis(), joinPoint.getSignature(), ex);
```
Comment (antonbakalets) [inline src/main/java/com/softserve/academy/spaced/repetition/logger/LogExceptionAspect.java]: This is why we use code review. Please be careful - you changed one word. Please revert to "...service".

Comment (vldmr1703) [reply src/main/java/com/softserve/academy/spaced/repetition/logger/LogExceptionAspect.java]: Sorry i'm just in a hurry.

**Pull request merged**